hack/generate-kcp-api.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+
+set -e
+
+if ! which kubectl-kcp; then
+  echo "kubectl-kcp required on path"
+  echo "you can install it with running:"
+  echo "    $ git clone https://github.com/kcp-dev/kcp && cd kcp && make install"
+  exit 1
+fi
+
+THIS_DIR="$(dirname "$(realpath "$0")")"
+CRD_DIR="$( realpath ${THIS_DIR}/../config/crd/bases)"
+KCP_API_DIR="$( realpath ${THIS_DIR}/../config/kcp)"
+
+KCP_API_SCHEMA_FILE_CURRENT="${KCP_API_DIR}/apiresourceschema_release.yaml"
+KCP_API_SCHEMA_FILE_NEW="${KCP_API_DIR}/apiresourceschema_release.yaml_new"
+cat << EOF > ${KCP_API_SCHEMA_FILE_NEW}
+# This file is generated from CRDs by ./hack/generate-kcp-api.sh script.
+# Please do not modify!
+
+EOF
+
+# APIResourceSchema is immutable so when we want to update something, we actually have to create new version.
+# Version is defined by this prefix, which is taken from date. This will allow us to do new version each minute, which
+# should be hopefully enough granularity :)
+PREFIX=$( TZ="Etc/UTC" date +%Y%m%d%H%M )
```

Comment:
As an alternative, you could use the output of md5sum for each CRD.

Although the date sounds good, this could become confusing really easily when two PRs, both modifying the CRDs and the APIResourceSchema, go to main. Depending on when the code was pushed (commit date) and the order followed to merge the PRs, the actual date could mean nothing (last PR merged could have a date previous to the other PR date and this is not fixed by a rebase). At least I think this would be the case... What do you think?

Reply:
This script is directly copied from the spi and has workstreams (just with spi/has replaced with release). I think we should keep it as is so we are like the other teams in hacbs

Reply:
This looks prone to failure but if everyone else is using the same script I guess consistency wins. To be revisited in case of problems.